manifest.go

```diff
@@ -0,0 +1,655 @@
+
+// DataFile is the interface for reading the information about a
+// given data file indicated by an entry in a manifest list.
+type DataFile interface {
+	// ContentType is the type of the content stored by the data file,
+	// either Data, Equality deletes, or Position deletes. All v1 files
+	// are Data files.
+	ContentType() ManifestEntryContent
+	// FilePath is the full URI for the file, complete with FS scheme.
+	FilePath() string
+	// FileFormat is the format of the data file, AVRO, Orc, or Parquet.
+	FileFormat() FileFormat
+	// Partition returns a mapping of field name to partition value for
+	// each of the partition spec's fields.
+	Partition() map[string]any
+	// Count returns the number of records in this file.
+	Count() int64
+	// FileSizeBytes is the total file size in bytes.
+	FileSizeBytes() int64
+	// ColumnSizes is a mapping from column id to the total size on disk
+	// of all regions that store the column. Does not include bytes
+	// necessary to read other columns, like footers. Map will be nil for
+	// row-oriented formats (avro).
+	ColumnSizes() map[int]int64
+	// ValueCounts is a mapping from column id to the number of values
+	// in the column, including null and NaN values.
+	ValueCounts() map[int]int64
+	// NullValueCounts is a mapping from column id to the number of
+	// null values in the column.
+	NullValueCounts() map[int]int64
+	// NaNValueCounts is a mapping from column id to the number of NaN
+	// values in the column.
+	NaNValueCounts() map[int]int64
+	// DistictValueCounts is a mapping from column id to the number of
+	// distinct values in the column. Distinct counts must be derived
+	// using values in the file by counting or using sketches, but not
+	// using methods like merging existing distinct counts.
+	DistinctValueCounts() map[int]int64
+	// LowerBoundValues is a mapping from column id to the lower bounded
+	// value of the column, serialized as binary. Each value in the column
+	// must be less than or requal to all non-null, non-NaN values in the
+	// column for the file.
+	LowerBoundValues() map[int][]byte
+	// UpperBoundValues is a mapping from column id to the upper bounded
+	// value of the column, serialized as binary. Each value in the column
+	// must be greater than or equal to all non-null, non-NaN values in
+	// the column for the file.
+	UpperBoundValues() map[int][]byte
+	// MetadataKey is implementation-specific key metadata for encryption.
+	MetadataKey() []byte
+	// Splits are the split offsets for the data file. For example,
+	// all row group offsets in a Parquet file. Must be sorted ascending.
+	Splits() []int64
+	// EqualityFieldIDs are used to determine row equality in equality
+	// delete files. It is required when the content type is
+	// EntryContentEqDeletes.
+	EqualityFieldIDs() []int
+	// SortOrderID returns the id representing the sort order for this
+	// file, or nil if there is no sort order.
+	SortOrderID() *int
+}
```

Comment:
does this need `DataSequenceNumber()` and `FileSequenceNumber()`? See also https://github.com/apache/iceberg/blob/ebce8538db20fd13859b6af841cf433d9423b53c/api/src/main/java/org/apache/iceberg/ContentFile.java#L130-L147

Reply:
Hmm, i'm reading through the Java code and the best I can see is that the `ContentFile` interface wraps around the entire `ManifestEntry` objects and essentially surfaces the individual fields of the data file to the same level as the manifest entry information (which includes the `DataSequenceNumber` and `FileSequenceNumber`. But if you look at the spec (https://iceberg.apache.org/spec/#manifests) those values exist in the `manifest_entry` struct, not the `data_file` struct, which is why they aren't exposed here at this level. 

Looking at the Builder for data files in the Java code (https://github.com/apache/iceberg/blob/d6bc248adb67de74e31dcb9c0af43fef68853d59/core/src/main/java/org/apache/iceberg/DataFiles.java) it also doesn't allow setting the sequence number values down there. You can see both the Data Sequence number and File Sequence Number in the `ManifestEntry` interface in the Go code.